pkg/config/creds.go

```diff
@@ -18,6 +19,10 @@ type Creds struct {
 	AppKey string
 }
 
+const credsKey = "creds"
+
+var credsCache sync.Map
```

Comment:
can we manage the credsCache in a struct. instead of a global env var? 

```suggestion
type CredentialManager struct {
    Cache sync.Map
}

func NewCredentialManager() *CredentialManager {
    return &CredentialManager{}
}
```

Reply:
The idea will be to create the `CredentialManager` in the `main()` func and pass it to the others components.

Reply:
`credsCache` is a local variable and isn't accesses by any other component, it's just used by `GetCredentials`. Not sure I understand the benefit of creating and passing a `CredentialManager` struct to the other components in main? we will end up with multiple/identical structs instantiated while a public util `GetCredentials` can be enough imho

Reply:
for example during unittest. 2 tests running in // will use the same cache.

Reply:
Yes, I applied your suggestion. Thanks!